### PR TITLE
User prefs

### PR DESCRIPTION
**Pull request by bridiver**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Ran `git rebase -i` to squash commits if needed.

requires https://github.com/brave/electron/pull/22

cc @bbondy @diracdeltas

### CODE REVIEW

File: js/state/contentSettings.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  ]
+}
+
+const getContentSettingsFromSiteSettings = (appState) => {
+  let braveryDefaults = siteSettings.braveryDefaults(appState, appConfig)
+
+  let contentSettings = {
+    cookies: braveryDefaults.cookieControl === 'block3rdPartyCookie' ? getBlock3rdPartyStorage() : getAllowAllCookies(),
+    javascript: [],
+    canvasFingerprinting: [{
```
Comment (diracdeltas) [inline js/state/contentSettings.js]: If you change this setting globally in about:preferences, it doesn't take effect until browser restart.

Comment (bridiver) [inline js/state/contentSettings.js]: works in https://github.com/brave/browser-laptop/pull/2256
we should change canvas fingerprinting to a resource like the others so they all work consistently

Comment (diracdeltas) [inline js/state/contentSettings.js]: the site setting works immediately though

Comment (bridiver) [inline js/state/contentSettings.js]: all the site settings work the same way, but the other resources are updated globally with APP_SET_RESOURCE_ENABLED and canvas fp is updated with APP_CHANGE_SETTING

Comment (bridiver) [inline js/state/contentSettings.js]: if you try again in master it should work

Comment (diracdeltas) [inline js/state/contentSettings.js]: ++, posted last comment before seeing your comments

Comment (diracdeltas) [inline js/state/contentSettings.js]: follow-up bug: disabling shield does not override the global fingerprinting setting.
screen shot 2016-06-21 at 2 25 45 pm

Comment (bridiver) [inline js/state/contentSettings.js]: yea, that makes sense and would apply to ads as well. contentSettings translate the siteSettings into rules so I can use the activeSettings (which will eventually be replaced by contentSettings), but I can add an override for now

Comment (bridiver) [inline js/state/contentSettings.js]: fixed here https://github.com/brave/browser-laptop/commit/9751f28f10edf4acb08e45e600ca77336e38b4d7

**Pull request merged**